File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/TypeTemplateService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbTypeTemplate;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface TypeTemplateService {
    public PageResult search( int pageNum ,int pageSize ,TbTypeTemplate typeTemplate);

    void add(TbTypeTemplate typeTemplate);

    void update(TbTypeTemplate typeTemplate);

    TbTypeTemplate findOne(Long id);

    List<TbTypeTemplate> findAll();

    List<Map> findSpecList(Long id);
}

File: pinyougou_parent/pinyougou_shop_web/src/main/webapp/js/service/typeTemplateService.js
app.service("typeTemplateService",function($http){

    this.findSpecList=function(id){
        return $http.get("../typeTemplate/findSpecList/"+id);
    }
    this.findAll=function () {
        return $http.get("../typeTemplate/findAll");
    }

    this.search=function(pageNum,pageSize,searchEntity){
       //return $http.post("../typeTemplate/search?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);

        // restful风格
       return $http.post("../typeTemplate/search/"+pageNum+"/"+pageSize,searchEntity);
    }

    this.add=function (entity) {
        return $http.post("../typeTemplate/add",entity);
    }
    this.update=function (entity) {
        return $http.post("../typeTemplate/update",entity);
    }
    this.findOne=function (id) {
        return $http.get("../typeTemplate/findOne/"+id);
    }


})